polars/polars-core/src/series/iterator.rs

```diff
@@ -144,7 +144,7 @@ impl Series {
 
 pub struct SeriesIter<'a> {
     arr: &'a dyn Array,
-    dtype: &'a DataType,
+    pub dtype: &'a DataType,
```

Comment:
any reason why this is private? is it OK to make it public?

Reply:
No, users can change the `DataType` and then our `AnyValue` behavior will have undefinded behaviour.

Reply:
@MarcoGorelli this has to be private as this is unsound.